sympy/core/evalf.py

```diff
@@ -1473,7 +1473,11 @@ def evalf(self, n=15, subs=None, maxn=100, chop=False, strict=False, quad=None,
             result = evalf(self, prec + 4, options)
         except NotImplementedError:
             # Fall back to the ordinary evalf
-            v = self._eval_evalf(prec)
+            from sympy.core.basic import Basic
+            if isinstance(self, Basic) and subs is not None:  # issue 20291
```

Comment:
What is the purpose of this change? Won't `self` always be `Basic`?

Reply:
`EvalfMixin` is not an instance of `Basic`. I think all its children are instances of `Basic` but I am not 100% sure. That's why I added it in. Should I remove the `Basic` check?